js-api/src/chem.ts

```diff
@@ -429,35 +422,62 @@ export namespace chem {
+      this.changedSub?.unsubscribe();
+      const sketcherFunc = this.sketcherFunctions.find(e => e.friendlyName == sketcherName) ?? this.sketcherFunctions.find(e => e.friendlyName == DEFAULT_SKETCHER);
+      this.sketcher = await sketcherFunc!.apply();
+      this.host!.style.minWidth = '500px';
+      this.host!.style.minHeight = '400px';
+      this.host.appendChild(this.sketcher!.root);
+      await ui.tools.waitForElementInDom(this.root);
+      await this.sketcher!.init(this);
+      ui.setUpdateIndicator(this.host, false);      
+      this.changedSub = this.sketcher!.onChanged.subscribe((_: any) => {
+        this.onChanged.next(null);
+        for (let callback of this.listeners)
+          callback();
+        if (this.syncCurrentObject) {
+          const molFile = this.getMolFile();
+          if (!Sketcher.isEmptyMolfile(molFile))
+            grok.shell.o = SemanticValue.fromValueType(molFile, SEMTYPE.MOLECULE, UNITS.Molecule.MOLBLOCK);
+        }
+      });
     }
 
-    static addFavorite(molecule: string) {
-      const favorites = Sketcher.getFavorites();
-      Sketcher.checkDuplicatesAndAddToStorage(favorites, molecule, Sketcher.FAVORITES_KEY);
+    setSketcherWithExistingMolecule(){
+      const getMolecule = async () => {
+        return this._smiles === null ? this._molfile === null ? this._smarts === null ? '' :
+          await this.getSmarts() : this.getMolFile() : this.getSmiles();
+      }
+      getMolecule().then((molecule) => {
+        this.setSketcher(currentSketcher).then((_) => {              
+          this.setMolecule(molecule!, this._smarts !== null);
+        });
+      });
     }
 
-    static getRecent(): string[] {
-      return JSON.parse(localStorage.getItem(Sketcher.RECENT_KEY) ?? '[]');
-    }
+    static readonly FAVORITES_KEY = 'chem-molecule-favorites';
+    static readonly RECENT_KEY = 'chem-molecule-recent';
 
-    static addRecent(molecule: string) {
-      const recent = Sketcher.getRecent();
-      Sketcher.checkDuplicatesAndAddToStorage(recent, molecule, Sketcher.RECENT_KEY);
+    static getCollection(key: string): string[] {
+      return JSON.parse(localStorage.getItem(key) ?? '[]');
+    }
+    
+    static addToCollection(key: string, molecule: string) {
+      const molecules = Sketcher.getCollection(key);
+      Sketcher.checkDuplicatesAndAddToStorage(molecules, molecule, key);
     }
 
     static checkDuplicatesAndAddToStorage(storage: string[], molecule: string, localStorageKey: string) {
       let mol1: any;
       function moleculesEqual(rdkitModule: any, mol1: any, molfile2: string): boolean {
         let mol2;
         try {
-          mol2 = rdkitModule.get_mol(molfile2);
+          mol2 = rdkitModule.get_mol(molfile2); //in case local storage contained malformed molecules wrap in try
         } catch (e: any) {
-          return false;
+          return false; 
         }
         const result = mol1.get_smiles() === mol2.get_smiles();
         mol2.delete();
```

Comment:
What if mol2 is null because of the error on line 478? You probably can do mol2?.delete()